common/openshift_ops.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from kubernetes import config
+from openshift.dynamic import DynamicClient
+
+log = logging.getLogger(__name__)
+
+
+class Ocp(object):
+    """Class which contains various utility functions for interacting
+    with OpenShift
+
+    """
+    def __init__(self):
+
+        k8s_client = config.new_client_from_config()
+        dyn_client = DynamicClient(k8s_client)
+
+        self.v1_service_list = dyn_client.resources.get(api_version='v1',
+                                                        kind='ServiceList')
+        self.v1_projects = dyn_client.resources.get(
+            api_version='project.openshift.io/v1', kind='Project')
+
+    def get_projects(self):
+        """Gets all the projects in the cluster.
+
+        Returns:
+            list: List of projects
+            NoneType: on failure
+        """
+        projects = []
+        try:
+            ret = self.v1_projects.get()
+            for each in ret.items:
+                projects.append(each.metadata.name)
+            return projects
+        except Exception as err:
+            log.error("Unable to list the projects. Error: %s", err)
+            return None
+
+    def get_services(self):
+        """Gets all the services in the cluster.
+
+        Returns:
+            dict: Dictionary of services, key represents the namespace
+                  and value represents the services
+            NoneType: on failure
+        """
+        services = {}
+        try:
+            ret = self.v1_service_list.get()
+            for each in ret.items:
+                if each.metadata.namespace in services:
+                    services[each.metadata.namespace].\
+                        append(each.metadata.name)
```

Comment:
again, you can convert to list comprehension

Reply:
> again, you can convert to list comprehension

Used Dictionary Comprehension